firmware/ethereum.c

```diff
@@ -568,7 +568,11 @@ void ethereum_signing_init(EthereumSignTx *msg, const HDNode *node)
 		rlp_length += rlp_calculate_length(1, tx_type);
 	}
 	if (chain_id) {
-		rlp_length += rlp_calculate_length(1, chain_id);
+		int length = 0;
+		if (msg->has_chain_id) {
```

Comment:
This is already in an `if (chain_id)`.  I think this nested `if` isn't necessary or even hurting.